source/merge.ts

```diff
@@ -27,10 +27,10 @@ export default function merge(target: Options, ...sources: Options[]): Options {
 		}
 	}
 
-	return target;
+	return target as Target & Source;
```

Comment:
This is needed since we don't have a temporary variable that stores the merged objects, but reuse the target passed in.